lib/rubocop/cop/style/file_name.rb

```diff
@@ -125,11 +125,18 @@ def to_namespace(path)
           # We can't assume that the working directory, or any other, is the
           # "starting point" to build a namespace
           start = %w(lib spec test src)
-          if components.find { |c| start.include?(c) }
-            components = components.drop_while { |c| !start.include?(c) }
-            components.drop(1).map { |fn| to_module_name(fn) }
-          else
+          start_index = nil
```

Comment:
I'd add a few comments to this code, as it's not really apparent why we're doing what we're doing.